palantirnet/search_api_federated_solr
# UMUS-115: react-ify mobile filters js

JIRA: [UMUS-115](https://palantir.atlassian.net/browse/UMUS-115)

This PR moves the search app mobile filter functionality out of jQuery and into reactJS.

### To prepare to test (in dev build)
1. Checkout this branch in `src/search_api_federated_solr`
1. CD into the search app `cd react-search`
1. Install dependencies `yarn install`
1. Run the dev task: `yarn start`
1. Observe a new browser window opens and the search app loads

### To confirm the filters form can be expanded / hidden in small viewport
1. Open your dev tools
1. Emulate a viewport less than 900px wide
1. Refresh the page
    1. Note: there was no event listener for window resize in legacy functionality, so I did not add any
1. Observe the filters are collapsed by default
1. Search for 'bones'
1. Expand the filters
1. Interact with them to ensure they still function as expected

### To confirm the filters still work in large viewport
1. Close your devtools
1. Refresh the page
    1. Note: there was no event listener for window resize in legacy functionality, so I did not add any
1. Search for 'bones'
1. Interact with sidebar filters, confirm functionality unchanged

### Screenshot
![screen recording 2018-05-07 at 02 22 pm](https://user-images.githubusercontent.com/3279883/39717734-2793297e-5202-11e8-92ac-87e2a5516c1f.gif)

### Note (for future ticket?)
I wonder if, upon clicking the clear all filters button, we shouldn't keep the original search query?  Seems strange to me to remove that as well.
